force-app/main/default/classes/OpportunityUtilities.cls

```diff
@@ -0,0 +1,259 @@
+/*
+ * Additional methods needed for the OpportunityTriggerHandler class
+ * 
+ * @author Patrick Sees
+*/
+
+
+public with sharing class OpportunityUtilities {
+
+    public static Boolean isStageChangeHandled = false;
+
+    public static Boolean  isPrimaryContactChangedAlready = false;
+
+
+    public static void defaultNewCustomers(List<Opportunity> oppys){
+
+        for (Opportunity opp : oppys) {
+            if (opp.Type == null){
+                opp.Type = 'New Customer';
+            }        
+        }        
+    }            
+
+    public static void preventClosedOppyDelete(List<Opportunity> oppys){
+
+    // Prevent deletion of closed Opportunities
+
+        for (Opportunity oldOpp : oppys){
+            if (oldOpp.IsClosed){
+                oldOpp.addError('Cannot delete closed opportunity');
+            }
+        }
+    }
+
+    public static void opportunityMustBeFiveThousand(List<Opportunity> oppys){
+
+    // When an opportunity is updated validate that the amount is greater than 5000.
+
+        for (Opportunity opp : oppys) {
+            if(opp.Amount < 5000){
+                opp.addError('Opportunity amount must be greater than 5000');
+            }
+        }
+    }
+
+
+    public static void cantDeleteClosedWonBanking(List<Opportunity> oppys){
+        
+    // When an opportunity is deleted prevent the deletion of a closed won opportunity if the account industry is 'Banking'.
+        List<Id> accIdList = new List<Id>();
+        for (Opportunity oldOpp : oppys) {
+            accIdList.add(oldOpp.AccountId);
+        }
+
+        Map<Id, Account> accounts = new Map<Id, Account>([SELECT Id, Industry FROM Account WHERE Id IN:accIdList]);
+
+        for (Opportunity opp : oppys) {
+            if (opp.Account.Industry == 'Banking' && opp.StageName == 'Closed Won')  {
```

Comment:
You would not have access to Account.Industry here because it is from a trigger and those fields would not be included. It does not matter here though because the requirements overlap and anything closed won cannot be deleted